### DISCUSSION TITLE
Recurse symlinked directory?

### DISCUSSION THREAD

**Discussion by oschrenk** [twpayne/chezmoi#2070] [General]
I'm in the process of migrating my dotfiles from a self-created symlink-ed solution to chezmoi.

When I try to add symlinked directories, it feels like the recursive loop is not respected.

This is the "outside" data I want to add to chezmoy
```
# symlink to directory
ls -la ~/.tldr
lrwxr-xr-x  1 oliver  staff  36 May 27  2020 /Users/oliver/.tldr -> /Users/oliver/.tilde/dotfiles/.tldr 

# contents of directory
ls ~/.tldr/
chrome.md  lsof.md    tar.md    

# adding directory to chezmoi
chezmoi add --follow --recursive "$HOME/.tldr"

```

But when I check the contents of the now tracked content in chezmoi:

```
ls -la ~/.local/share/chezmoi/home/dot_tldr/
total 0
drwxr-xr-x   3 oliver  staff   96 May 16 08:37 .
drwxr-xr-x  21 oliver  staff  672 May 16 08:37 ..
-rw-r--r--   1 oliver  staff    0 May 16 08:37 .keep
```

chezmoi only creates a `.keep` file.

It seems to me that `--follow` does not recurse the directory. Other then manually copying the files, is there a flag I am maybe missing?

**Comment by oschrenk** [comment]
I realized that it might be a design decision and I am good with it.

I ended up migrating my dotfiles in chunks manually and it forced me to not only clean up but also understand the concepts of chezmoi. I ran into little things like not needing the gitignore files, or needing to set certain chattr on source files. All those little learnings might have been lost if I wouldn't have done it manually or semi-automatic (i still relied on some for loops)

I consider this discussion closed.

**Reply by twpayne** [reply]
Great! Thank you very much for the follow-up.
